examples/keyword-extraction-api.ipynb

```diff
@@ -0,0 +1,615 @@
+    "        \"Marine Expedition Finds New Species - Norwegian scientists who explored the deep waters of the Atlantic Ocean said Thursday their findings #151; including what appear to be new species of fish and squid #151; could be used to protect marine ecosystems worldwide.: nan\"\n",
+    "    ]\n",
+    "})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "OyGuHllZllct",
+   "metadata": {
+    "id": "OyGuHllZllct"
+   },
+   "source": [
+    "## Adaptive Inference"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "64c345aa-b6a7-4770-8368-b290e9e799dc",
+   "metadata": {
+    "id": "6-MZlfXAoiNv"
+   },
+   "source": [
+    "To run the inference job, we’ll follow three simple steps:\n",
+    "1. **Create the inference file -** we’ll create a file containing the requests to be processed by the model.\n",
+    "2. **Upload the inference file -** once the file is ready, we’ll upload it to the kluster.ai platform using the API, where it will be queued for processing.\n",
+    "3. **Start the job -** after the upload, we’ll trigger the job to process the data.\n",
+    "\n",
+    "Everything is preconfigured for you—just execute the cells below to see it all in action!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "Ew-R24Ltp5EW",
+   "metadata": {
+    "id": "Ew-R24Ltp5EW"
+   },
+   "source": [
+    "### Create the Adaptive Inference file"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "qS4JXT52wGJ-",
+   "metadata": {
+    "id": "qS4JXT52wGJ-"
+   },
+   "source": [
+    "This example uses the `klusterai/Meta-Llama-3.1-70B-Instruct-Turbo` model. If you’d prefer to use a different model, you can easily modify the model name in the next cell. For a full list of supported models, please check our <a href=\"https://docs.kluster.ai/getting-started/#list-supported-models\" target=\"_blank\">documentation</a>."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "id": "fVtwyqZ_nEq7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_inference_file(df):\n",
+    "    inference_list = []\n",
+    "    for index, row in df.iterrows():\n",
+    "        content = row['text']\n",
+    "        \n",
+    "        request = {\n",
+    "            \"custom_id\": f\"keyword_extraction-{index}\",\n",
+    "            \"method\": \"POST\",\n",
+    "            \"url\": \"/v1/chat/completions\",\n",
+    "            \"body\": {\n",
+    "                \"model\": \"klusterai/Meta-Llama-3.1-70B-Instruct-Turbo\",\n",
+    "                \"temperature\": 0.5,\n",
+    "                \"response_format\": {\"type\": \"json_object\"},\n",
+    "                \"messages\": [\n",
+    "                    {\"role\": \"system\", \"content\": 'Extract up to 5 relevant keywords from the given text. Provide only the keywords between double quotes and separated by commas.'},\n",
```

Comment:
In the prompt, define the JSON that the model should output:
{
  "document_id": "unique_id",
  "original_text": "the input text",
  "keywords": [
    {
      "keyword": "extracted keyword",
      "confidence": 0.00,
      "frequency": 0
    }
  ]
}